ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -0,0 +1,59 @@
+# Testing Function
+# global
+from hypothesis import strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+import ivy_tests.test_ivy.test_frontends.test_numpy.helpers as np_frontend_helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+import ivy
+
+
+@handle_frontend_test(
+    fn_tree="numpy.diagonal",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=1,
+        shared_dtype=True,
+        min_num_dims=2,
+        min_axis=-2,
+        max_axis=1,
+    ),
```

Comment:
And, then add `min_axes_size=2` and `max_axes_size=2`. This will ensure that the returned axis is a tuple with 2 elements.

Reply:
now just using
`fn_tree="numpy.diagonal",
    dtype_x_axis=helpers.dtype_values_axis(
        available_dtypes=helpers.get_dtypes("numeric"),
        min_axes_size=2,
        max_axes_size=2,`
further 
`input_dtype, x, axis  = dtype_x_axis`
still axis has a single value and the axis is also picking any integer value (maybe that would be right according to backend work).

Reply:
There was a bug in `get_axis` that caused this problem that's hopefully now fixed. Please merge the current master to the PR and try again with 
`
        min_num_dims=2,
        min_axes_size=2,
        max_axes_size=2,
        valid_axis=True,
`

Reply:
You forgot `min_num_dims=2`